mtools/mlaunch/mlaunch.py

```diff
@@ -1915,7 +1915,7 @@ def _wait_for_primary(self):
                   for x in self.config_docs[self.args['name']]['members']])
         rs_name = self.config_docs[self.args['name']]['_id']
         mrsc = self.client(hosts, replicaSet=rs_name,
-                           serverSelectionTimeoutMS=30000)
+                           serverSelectionTimeoutMS=30000, directConnection=False)
```

Comment:
It would probably be more robust to make this change in `MongoConnection` instead:
```
    def __init__(self, *args, **kwargs):
-        kwargs.setdefault('directConnection', True)
+        if not kwargs.get('replicaSet'):
+            kwargs.setdefault('directConnection', True)
        kwargs.setdefault('serverSelectionTimeoutMS', 1)
```